packages/kit/src/runtime/server/index.js

```diff
@@ -309,7 +310,8 @@ export async function respond(request, options, state) {
 			// so we need to make an actual HTTP request
 			return await fetch(request);
 		} catch (e) {
-			const error = coalesce_to_error(e);
+			// HttpError can come from endpoint - TODO should it be handled there instead?
+			const error = e instanceof HttpError ? e : coalesce_to_error(e);
```

Comment:
do we still need `coalesce_to_error`? i've always been a bit fuzzy on the reason for that function, feels like we can probably get rid of it once and for all in this PR

Reply:
Good question honestly haven't looked up where else it's used.

Reply:
it's used in a few different places, but all of them are a bit weird. though honestly it might be easier to disentangle them separately